src/queries.rs

```diff
@@ -758,17 +769,18 @@ impl<'a> VintedWrapper<'a> {
 
         match json.status() {
             StatusCode::OK => {
-                let response_text = json.text().await?;
-                let result: Result<AdvancedItems, serde_json::Error> =
-                    serde_json::from_str(&response_text);
-
-                if result.is_err() {
-                    log::error!("{}", &response_text)
+                let raw_json = json.text().await?;
+                match serde_json::from_str::<AdvancedItems>(&raw_json) {
```

Comment:
Same for here

Deserialize as json directly.

Reply:
[make Json transparent in the error](https://github.com/ThalosES/vinted-rs/pull/118/commits/d45f800f09339bacc15584715d32633c0024ec49)